action.yml

```diff
@@ -0,0 +1,159 @@
+name: Code Engine GitHub Action
+description: Github action to interact with IBM Cloud Code Engine deploy Apps, Jobs and Functions
+author: IBM
+
+branding:
+  icon: cloud
+  color: blue
+
+inputs:
+  api-key:
+    description: IAM API Key used to log into the IBM Cloud
+    required: true
+
+  resource-group:
+    description: An IBM Cloud Resource Group is a logical container used to organize and manage related cloud resources
+    required: false
+    default: Default
+
+  region:
+    description: The Region refers to a geographical area where the project is are located
+    required: true
+
+  project:
+    description: A Code Engine Project Is the grouping of your Apps, Functions and Jobs
+    required: true
+
+# App, Function or Job Specific inputs
+  entity:
+    description: The type of entity that should be deployed [App, Function, Job]
+    required: true
+
+  name: 
+    description: Name of the App, Function or Job
+    required: true
+  
+  runtime:
+    description: Runtime used for the Function only required for function
+    required: false
+
+  build-source:
+    description: path to the directory containing the source code 
+    required: false
+    default: .
+
+  cpu:
+    description: CPU configuration set for the entity. If not set default Code Engine values are used. 
+    required: false
+
+  memory:
+    description: Memory configuration set for the entity. If not set default Code Engine values are used.
+    required: false
+
+# actual action code 
+runs:
+  using: composite
+
+  steps:
+    # Default steps required
+    - name: Install IBM Cloud CLI
+      shell: bash
+      run: curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+    
+    - name: Login to IBM Cloud And target Resouce Groupe and Region
+      shell: bash
+      run: ibmcloud login --apikey ${{ inputs.api-key }} -r ${{ inputs.region }} -g ${{ inputs.resource-group }}
+
+    - name: Install Code Engine Plugin
+      shell: bash
+      run: ibmcloud plugin install code-engine
+
+    # Select the project using Name or ID
+    # If the project doesn`t exist the project is created with the provided Name
+    - name: Select Code Engine Project
+      shell: bash
+      run: | 
+        if ibmcloud ce project select --name ${{ inputs.project }} || ibmcloud ce project select --id ${{ inputs.project }} ; then
```

Comment:
encapsulate the name with quotes to enable like 'My project'

Reply:
fixed